aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -511,15 +540,28 @@ static Tensor squeeze_multiple(const Tensor& self, IntArrayRef dims) {
 }
 
 static Tensor& logsumexp_out_impl(Tensor& result, const Tensor& self, IntArrayRef dims, bool keepdim) {
+  TORCH_CHECK(!at::isIntegralType(result.scalar_type(), /*includeBool=*/true),
+    "logsumexp requires a floating output type, but got ",
+    result.scalar_type());
+
+  // Casts integral inputs to the appropriate (floating) type
+  Tensor cast_self;
+  if (at::isIntegralType(self.scalar_type(), /*includeBool=*/true)) {
```

Comment:
Fold this conditional into the ternary operation below

Reply:
if all of the constituents support upcasting, can't you just...do nothing here?

Reply:
Offline conversation: let's comment that this should be removed once the unary floating ufunc PR is in, and be sure to ward people away from implementing a similar excessive cast for "composite functions" like logsumexp.